src/sound/instance.js

```diff
@@ -573,15 +598,51 @@ class SoundInstance extends EventHandler {
     }
 
     /**
-     * Begins playback of sound. If the sound is not loaded this will return false. If the sound is
-     * already playing this will restart the sound.
+     * Attempt to begin playback the sound.
+     * If the AudioContext is suspended, the audio will only start once it's resumed.
+     * If the sound is already playing, this will restart the sound.
      *
-     * @returns {boolean} True if the sound was started.
+     * @returns {boolean} True if the sound was started immediately.
      */
     play() {
         if (this._state !== STATE_STOPPED) {
             this.stop();
         }
+        // set state to playing
+        this._state = STATE_PLAYING;
+        // no need for this anymore
+        this._playWhenLoaded = false;
```

Comment:
Looks like `slot.js` still references this member [here](https://github.com/playcanvas/engine/blob/main/src/framework/components/sound/slot.js#L167). If this member is no longer used, shouldn't it just be removed?

Reply:
Are you referencing the comment `no need for this anymore`? If so, these 4 lines of code were already in code, but I had to move them, and I think that the `no need for this anymore` is about no longer needing for `_playWhenLoaded` to be true. The way that property is used is weird, but I'd prefer not to change too much of that in this PR (unless you see another issue).